### PR TITLE
Replaced included default player actions with custom player actions

### LINKED ISSUE #139: Custom actions asset for EventSystem
## Description
By default the EventSystem object uses DefaultInputActions. It should use our custom Controls action asset found in Scripts/Input. Controls should be modified to mimic functionality of DefaultInputActions. This will allow controls to be changed without updating two separate action assets.

## Acceptance Criteria

This story will be finished when the EventSystem actions asset has been updated to use our Controls action asset.

### PR DESCRIPTION
**Pull request by ViMaSter**
## Summary
This PR removes all references in our project to the `DefaultPlayerActions` that are included as an example inside the `Input System` package and instead uses the `PlayerActions` instance at `/Assets/Scripts/Input/Control.cs` were using for the rest of the game already.

### Unused UI input system actions
Unity's UI Input system consists of some behaviors we're currently not using. As we're not using them, I've deliberately left them unassigned:
- Middle Click
- Right Click
- Scroll Wheel
- Tracked Position and Orientation (not to be confused with the mouse position called `Point`, that's assigned; think a phone's gyro sensor or a Wiimote)

image

### `DirectionalButtons` vs. individual `Left`, `Right`, `Up`, `Down` actions
Our `Controls` had both `DirectionalButtons` **and** individual `Left`, `Right`, `Up`, `Down` actions. I've removed the individual actions as they a) seemed unused and b) can easily be accessed by listening to changes of the `DirectionalButtons` axis.  
It now looks like this:  
image

### `X` and `Enter`
`X` and `Enter` can now be used interchangeably. I haven't really seen the need to separate the two, but if we want to split "Dialogue actions" and "UI actions" later on, it's something we can always do.

## Testing instructions
1. Run any scene of the game
2. Make sure all actions previously executable are still executable; aside from `X` and `Enter` now able to be used interchangeably, no keybinds should have changed
3. Verify the EditMode and PlayMode tests still succeed

## Additional information

- `[x]` Changes UI
  - `X` and `Enter` key can now be used interchangeably 
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
  - Closes #139

### CODE REVIEW

File: unity-ggjj/Assets/Scenes/Inky-TestScene.unity [hunk 1]
```diff
@@ -130,83 +130,68 @@ PrefabInstance:
   m_Modification:
     m_TransformParent: {fileID: 749697830}
     m_Modifications:
-    - target: {fileID: 6276402831480686355, guid: ba051a1745ca9df40845054bf2e42823,
-        type: 3}
+    - target: {fileID: 6276402831480686355, guid: ba051a1745ca9df40845054bf2e42823, type: 3}
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scenes/Inky-TestScene.unity]: Unrelated to this line, but the input actions needs to be updated in the MainMenu scene

**Review by PhoebeMitchell — DISMISSED** [review]
Thanks for taking on this issue, because I really didn't wanna do it :D

File: unity-ggjj/Assets/Scenes/Inky-TestScene.unity (hunk 1 above)
Comment (ViMaSter) [reply unity-ggjj/Assets/Scenes/Inky-TestScene.unity]: ...and actually not just there, it turns out:
image
Well spotted! Shoud've done that search in the first place, so thanks for that find! Updated those scenes in https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/155/commits/4ddc4908b44115ef7a0333b15636bf6ecf6117b9.

**Review by notexplosive — DISMISSED** [review]
I don't really understand the changes (looks like Unity nonsense). But if it works it works, I'll take it!

**Review by lyndsiWilliams — CHANGES_REQUESTED** [review]
These tests are failing consistently:
image

**Comment by ViMaSter** [comment]
@lyndsiWilliams Hmm... I was actually able to run them locally.
image
Can you check if this is related to the virtual input device limit issue we ran into on the other branch too?

If the logs contain something along the lines of `NotSupportedException: Control count per binding cannot exeed byte.MaxValue=255` restarting the Unity Editor should fix it.

---

Also feel free to always attach the full logs any of the failed tests. If the comment is getting too big, a link to a secret gist at http://gist.github.com/ works too. : D (Just paste the full log, click `Create secret gist` from the dropdown button at the bottom right and paste the URL you land on here!)
image
https://gist.github.com/ViMaSter/66fa237b0f1354cf5cc271a6d2e5476e

**Comment by lyndsiWilliams** [comment]
> @lyndsiWilliams Hmm... I was actually able to run them locally. image Can you check if this is related to the virtual input device limit issue we ran into on the other branch too?
> 
> If the logs contain something along the lines of `NotSupportedException: Control count per binding cannot exeed byte.MaxValue=255` restarting the Unity Editor should fix it.
> 
> Also feel free to always attach the full logs any of the failed tests. If the comment is getting too big, a link to a secret gist at http://gist.github.com/ works too. : D (Just paste the full log, click `Create secret gist` from the dropdown button at the bottom right and paste the URL you land on here!) image https://gist.github.com/ViMaSter/66fa237b0f1354cf5cc271a6d2e5476e

Very strange indeed, it doesn't seem to be the same errors that we were seeing before. Here's a gist with the errors I'm seeing on my side: https://gist.github.com/lyndsiWilliams/f0ef510635f5b4c6532611a70610cd1d

Also, I didn't know github gists were a thing, that's handy! 😮

**Comment by ViMaSter** [comment]
@lyndsiWilliams Since most of the failing tests are mouse-related, are you running the tests and then focussing another window or running them in the background? Unity's Input System can fail if you run all tests in succession but the Unity Editor isn't focussed.

If that also doesn't work, I've noticed there's logs about missing GameObjects and Components; can you actually run the `AppearingDialogueController - Test Scene` by hand, press `Esc` and open the menu? Or are there any errors inside the Console (`Window -> General -> Console`) when you try and do that?
image

@notexplosive / @PhoebeMitchell Can anyone of you reproduce this as well?

**Review by PhoebeMitchell — APPROVED** [review]
Tests run fine on my end

**Review by lyndsiWilliams — APPROVED** [review]
Sorry for the holdup on this - all the tests run on my end now! Not sure what the issue was before 🤷

**Pull request merged**